src/interpreter/lib/std.ts

```diff
@@ -638,41 +638,59 @@ export const std: Record<string, Value> = {
 			if (immediate.value) opts.call(callback, []);
 		}
 
-		const id = setInterval(() => {
-			opts.topCall(callback, []);
-		}, interval.value);
-
-		const abortHandler = (): void => {
+		let id: ReturnType<typeof setInterval>;
+
+		const start = (): void => {
+			id = setInterval(() => {
+				opts.topCall(callback, []);
+			}, interval.value);
+			opts.registerAbortHandler(stop);
+			opts.registerPauseHandler(stop);
+			opts.unregisterUnpauseHandler(start);
+		};
+		const stop = (): void => {
 			clearInterval(id);
+			opts.unregisterAbortHandler(stop);
+			opts.unregisterPauseHandler(stop);
+			opts.registerUnpauseHandler(start);
 		};
 
-		opts.registerAbortHandler(abortHandler);
+		start();
 
 		// stopper
 		return FN_NATIVE(([], opts) => {
-			clearInterval(id);
-			opts.unregisterAbortHandler(abortHandler);
+			stop();
+			opts.unregisterUnpauseHandler(start);
 		});
 	}),
 
 	'Async:timeout': FN_NATIVE(async ([delay, callback], opts) => {
 		assertNumber(delay);
 		assertFunction(callback);
 
-		const id = setTimeout(() => {
-			opts.topCall(callback, []);
-		}, delay.value);
+		let id: ReturnType<typeof setInterval>;
 
-		const abortHandler = (): void => {
+		const start = (): void => {
+			id = setTimeout(() => {
+				opts.topCall(callback, []);
```

Comment:
setTimeoutが完了してもハンドラが残ってると、時間経過したあとに`interpreter.pause()`→`interpreter.unpause()`と実行されるとコールバック関数が二回呼び出されると思います。

```suggestion
				opts.topCall(callback, []);
				opts.unregisterAbortHandler(stop);
				opts.unregisterPauseHandler(stop);
```